pom.xml

```diff
@@ -120,22 +120,25 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
 
-        <beam.version>2.19.0</beam.version>
+        <beam.version>2.20.0</beam.version>
         <postgresql.version>42.2.12</postgresql.version>
         <mysql.version>8.0.19</mysql.version>
         <socket-factory.version>1.0.14</socket-factory.version>
 
         <!--Ensure Beam SDK compatibility-->
         <!--https://github.com/apache/beam/blob/master/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy-->
-        <jackson.version>2.9.10</jackson.version>
+        <guava.version>28.2-android</guava.version>
```

Comment:
🙄

Reply:
Well, it is the same as the one on Beam. :/

Reply:
Beam uses guava 25.1-jre as direct declared dependency. This is actually pulled by a transitive dep.

Reply:
Right.. But I learned that is misleading. GRPC deps need a newer Guava version. Also I learned that `-android` is targeted for android/jdk7, and `-jre` to jdk8. Hence I believe `-jre` is the best choice for Beam related projects (no need for jdk7).

Reply:
yes gRPC is one of the deps that brings higher version of guava, however Beam brings a lot more transitive deps that still use guava below 25 (some of them still on 20). Being on 25.1-jre is actually important in Beam's case because it's the highest version that it's still compatible with version 20. So yeah Beam is playing safe because guava bin compat was kinda of introduced only in 21 and because of that gRPC can work™  with older guava.

Guava is really a mess and I guess In dbeam case you can get away 😄